bastion-executor/src/load_balancer.rs

```diff
@@ -47,46 +51,102 @@ impl LoadBalancer {
     }
 }
 
+/// Maximum number of core supported by mordern computers.
+const MAX_CORE: usize = 40;
+
 ///
 /// Holding all statistics related to the run queue
 ///
 /// Contains:
-/// * Global run queue size
 /// * Mean level of processes in the run queues
 /// * SMP queue distributions
-#[derive(Clone, Debug)]
 pub struct Stats {
-    pub(crate) global_run_queue: usize,
-    pub(crate) mean_level: usize,
-    pub(crate) smp_queues: FxHashMap<usize, usize>,
+    smp_load: [AtomicUsize;MAX_CORE],
+    mean_level: AtomicUsize,
 }
 
-unsafe impl Send for Stats {}
-unsafe impl Sync for Stats {}
+impl Stats{
+    /// new returns LockLessStats
+    pub fn new(num_cores: usize)  -> Stats{
+
+        let smp_load: [AtomicUsize; MAX_CORE] = {
+            let mut data: [MaybeUninit<AtomicUsize>; MAX_CORE] = unsafe{
+                MaybeUninit::uninit().assume_init()
+            };
+            let mut i = 0;
+            while i < MAX_CORE{
+                if i < num_cores {
+                    unsafe{std::ptr::write(data[i].as_mut_ptr(),AtomicUsize::new(0));}
+                    i = i+1; 
+                    continue;   
+                }
+                // MAX is for unused slot.
+                unsafe{std::ptr::write(data[i].as_mut_ptr(),AtomicUsize::new(usize::MAX));}
```

Comment:
This doesn't look like it's formatted. Let's format this with cargo fmt.

Reply:
Done